### PR TITLE
reference/transactions: add read committed isolation level

### PR DESCRIPTION
**Pull request by ran-huang**
### What is changed, added or deleted? 

Add read committed isolation level for the transaction chapter.

### What is the related PR or file link(s)? 

- [ ] Reference link(s):
- [x] This PR is to align with: [pingcap/docs-cn#2206](https://github.com/pingcap/docs-cn/pull/2206)
- [ ] N/A (not applicable)

### Which TiDB version(s) does your changes apply to? 

- [ ] All active versions: dev, v3.0, v2.1, v3.1
- [x] dev (the latest development version)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] All active and inactive versions
- [ ] N/A (not applicable)

**Note:** If your changes apply to multiple TiDB versions, make sure you update the documents in the **corresponding** version folders such as "dev", "v3.0", "v2.1" and "v3.1" in this PR.

- [ ] Updated one version first. Will update other versions after I get two LGTMs.

### CODE REVIEW

File: dev/reference/transactions/transaction-isolation.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ TiDB implements Snapshot Isolation consistency, which it advertises as `REPEATAB
 
 TiDB uses the [Percolator transaction model](https://research.google.com/pubs/pub36726.html). A global read timestamp is obtained when the transaction is started, and a global commit timestamp is obtained when the transaction is committed. The execution order of transactions is confirmed based on the timestamps. To know more about the implementation of TiDB transaction model, see [MVCC in TiKV](https://pingcap.com/blog/2016-11-17-mvcc-in-tikv/).
 
-## Repeatable Read
+## Repeatable Read isolation level
```
Comment (jackysp) [inline dev/reference/transactions/transaction-isolation.md]:
```suggestion
## Repeatable Read Isolation Level
```
How about this?

File: dev/reference/transactions/transaction-isolation.md [hunk 2]
```diff
@@ -54,6 +54,16 @@ The Repeatable Read isolation level in TiDB differs from that in MySQL. The MySQ
 
 The MySQL Repeatable Read isolation level is not the snapshot isolation level. The consistency of MySQL Repeatable Read isolation level is weaker than both the snapshot isolation level and TiDB Repeatable Read isolation level.
 
+## Read Committed isolation level
+
+TiDB supports the Read Committed isolation level only in [Pessimistic Transaction Mode](/dev/reference/transactions/transaction-pessimistic.md). In the optimistic transaction mode, setting the transaction level to Read Committed will not take effect. Transactions will still use the Read Committed isolation level.
```
Comment (jackysp) [inline dev/reference/transactions/transaction-isolation.md]:
```suggestion
TiDB supports the Read Committed isolation level only in [Pessimistic Transaction Mode](/dev/reference/transactions/transaction-pessimistic.md). In the optimistic transaction mode, setting the transaction isolation level to Read Committed will not take effect. Transactions will still use the Repeatable Read isolation level.
```

Comment (ran-huang) [reply dev/reference/transactions/transaction-isolation.md]: Thanks! Already committed your suggestions.

File: dev/reference/transactions/transaction-isolation.md (hunk 1 above)
Comment (ran-huang) [reply dev/reference/transactions/transaction-isolation.md]: Thanks for pointing out!
I have some doubts about the capitalization of headings though. I checked the related chapter in [Google developer documentation style guide](https://developers.google.cn/style/capitalization.html), and it suggests only capitalizing the first word unless it's a proper noun.
Could you please kindly tell me is "isolation level" here a proper noun? 
If so, I will commit the suggestion.

Comment (jackysp) [reply dev/reference/transactions/transaction-isolation.md]: I refer to https://www.postgresql.org/docs/current/transaction-iso.html. And the all-caps may also be OK https://dev.mysql.com/doc/refman/8.0/en/innodb-transaction-isolation-levels.html#isolevel_repeatable-read

Comment (ran-huang) [reply dev/reference/transactions/transaction-isolation.md]: Thanks for the references! 
I'll commit the suggestion.

Comment (lilin90) [reply dev/reference/transactions/transaction-isolation.md]: @ran-huang Actually we follow this style: use capital letters for document titles and sentence-style for page headings. 😀You can take a look at our documentation.

Comment (ran-huang) [reply dev/reference/transactions/transaction-isolation.md]: @lilin90 Thanks for pointing out.
I am still a little confused though. I noticed that some of our docs use capital letters for second-level and third-level headings, e.g. [docs/README.md](https://github.com/pingcap/docs/blob/master/README.md), while others use sentence-style, e.g. [docs/data-migration-with-binary.md](https://github.com/pingcap/docs/blob/master/dev/how-to/deploy/data-migration-with-binary.md). 
Which style should I choose for this PR?

File: dev/reference/transactions/transaction-isolation.md [hunk 3]
```diff
@@ -54,6 +54,16 @@ The Repeatable Read isolation level in TiDB differs from that in MySQL. The MySQ
 
 The MySQL Repeatable Read isolation level is not the snapshot isolation level. The consistency of MySQL Repeatable Read isolation level is weaker than both the snapshot isolation level and TiDB Repeatable Read isolation level.
 
+## Read Committed isolation level
+
+TiDB supports the Read Committed isolation level only in [Pessimistic Transaction Mode](/dev/reference/transactions/transaction-pessimistic.md). In the optimistic transaction mode, setting the transaction isolation level to Read Committed will not take effect. Transactions will still use the Repeatable Read isolation level.
```
Comment (lilin90) [inline dev/reference/transactions/transaction-isolation.md]: Try to limit the use of "will" where possible.

```suggestion
TiDB supports the Read Committed isolation level only in the [Pessimistic Transaction Mode](/dev/reference/transactions/transaction-pessimistic.md). In the optimistic transaction mode, setting the transaction isolation level to Read Committed does not take effect. Transactions will still use the Repeatable Read isolation level.
```

File: dev/reference/transactions/transaction-isolation.md (hunk 1 above)
Comment (lilin90) [reply dev/reference/transactions/transaction-isolation.md]: @ran-huang This is because almost all lines in `README.md` are document titles.

Comment (ran-huang) [reply dev/reference/transactions/transaction-isolation.md]: @lilin90 Got it! Thank you for explaining in detail.😊

**Pull request merged**